drools-parser/src/main/java/org/drools/parser/DRLParserHelper.java

```diff
@@ -15,11 +16,8 @@ private DRLParserHelper() {
     }
 
     public static PackageDescr parse(String drl) {
-        return compilationUnitContext2PackageDescr(createParseTree(drl));
-    }
-
-    public static DRLParser.CompilationUnitContext createParseTree(String drl) {
-        return createDrlParser(drl).compilationUnit();
+        DRLParser drlParser = createDrlParser(drl);
```

Comment:
Hi @tkobayas Thanks for PR. There is a strange pattern going on here:
1. first, a `DRLParser` object is instantiated
2. then, a `PackageDescr` is created, returned, using a couple of the properties of the above `DRLParser`
3. finally, the `DRLParser` instance is thrown away.

Basically, instantiation of `DRLParser` here is needed only to retrieve a `CompilationUnit` and a `TokenStream` - but IMO we should avoid to instantiate objects only as "shortcut" to get a couple of objects out of them, and then throw them away. WDYT ?

Reply:
@gitgabrio  Sorry, I'm not understanding well.

`DRLParser` is indeed a class to generate a `CompilationUnitContext` and transform a `TokenStream`. (It's done during `drlParser.compilationUnit()`)

Is it a "shortcut"? It looks normal to me that `DRLParser` is discarded after the generation.

Could you kindly share an expected usage code?

Reply:
Hi @tkobayas Thanks for explanation!